graph-subscriptions-api/src/network_subgraph.rs

```diff
@@ -62,6 +63,21 @@ impl SubgraphDeployments {
             .get(deployment)
             .cloned()
     }
+    pub async fn deployment_subgraphs_res(
+        &self,
+        deployment: DeploymentId,
+    ) -> anyhow::Result<Vec<Subgraph>> {
+        let map = self.inputs.value().await.ok();
+        if map.is_none() {
+            return Ok(vec![]);
+        }
+        Ok(map
+            .unwrap()
+            .deployment_to_subgraphs
+            .get(&deployment)
+            .cloned()
+            .unwrap_or(vec![]))
+    }
```

Comment:
(more comments around this below)
This is probably a me having a lack of rust knowledge/experience hack, but using this approach, that returns a `Result<Vec<Subgraph>>` allowed me to pass it in to a `Futures` impl that I needed to use (see below) to get a flattened `Vec<Subgraph>` for the unique deployment qm hashes on the records stored in the db.

Reply:
I think it makes more sense to have one function that returns a vec, since in this case None is just treated as an empty vec anyway.
```rust
pub async fn deployment_subgraphs(&self, deployment: &DeploymentId) -> Vec<Subgraph> {
    let map = match self.inputs.value().await {
        std::result::Result::Ok(map) => map,
        Err(_) => return vec![],
    };
    map.deployment_to_subgraphs
        .get(deployment)
        .cloned()
        .unwrap_or_default()
}
```
Also, pattern matching should be preferred over checking `is_none` then `unwrap`-ing.

The combination of making the return type a vec and the `deployment` parameter a reference type also simplifies the code at the callsite:
```rust
let subgraphs = join_all(uniq_deployment_hashes.iter().map(|deployment| {
    schema_ctx
        .subgraph_deployments
        .deployment_subgraphs(&deployment)
}))
.await
.into_iter()
.flatten();
Ok(Some(
    subgraphs.skip(skip as usize).take(first as usize).collect(),
))
```
The reference type gets rid of the annoying capture dance I had to do to disambiguate between what's referenced and what's moved, since now both `deployment` and `schema_ctx` are captured as references in the closure.